include/deal.II/base/polynomial.h

```diff
@@ -87,7 +95,8 @@ namespace Polynomials
      * Return the value of this polynomial at the given point.
      *
      * This function uses the Horner scheme for numerical stability of the
-     * evaluation.
+     * evaluation for polynomials in the coefficient form or the product of
+     * terms involving the roots if that representation is used.
```

Comment:
The selection is done upon construction of the object. You either give it an array of coefficients or an array of roots. If polynomials are added or some other form that changes the root, the internal representation is switched to the coefficient form.